blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/UserImpl.java

```diff
@@ -149,25 +162,73 @@ private boolean isAdmin(){
 
     private Map<String, Boolean> getPipelinePermissions(){
         return ImmutableMap.of(
-                BluePipeline.CREATE_PERMISSION, Jenkins.getInstance().hasPermission(Item.CREATE),
-                BluePipeline.READ_PERMISSION, Jenkins.getInstance().hasPermission(Item.READ),
-                BluePipeline.START_PERMISSION, Jenkins.getInstance().hasPermission(Item.BUILD),
-                BluePipeline.STOP_PERMISSION, Jenkins.getInstance().hasPermission(Item.CANCEL),
-                BluePipeline.CONFIGURE_PERMISSION, Jenkins.getInstance().hasPermission(Item.CONFIGURE)
+                BluePipeline.CREATE_PERMISSION, orgBase.hasPermission(Item.CREATE),
+                BluePipeline.READ_PERMISSION, orgBase.hasPermission(Item.READ),
+                BluePipeline.START_PERMISSION, orgBase.hasPermission(Item.BUILD),
+                BluePipeline.STOP_PERMISSION, orgBase.hasPermission(Item.CANCEL),
+                BluePipeline.CONFIGURE_PERMISSION, orgBase.hasPermission(Item.CONFIGURE)
         );
     }
 
     private Map<String, Boolean> getCredentialPermissions(){
         return ImmutableMap.of(
-                CREDENTIAL_CREATE_PERMISSION, Jenkins.getInstance().hasPermission(CredentialsProvider.CREATE),
-                CREDENTIAL_VIEW_PERMISSION, Jenkins.getInstance().hasPermission(CredentialsProvider.VIEW),
-                CREDENTIAL_DELETE_PERMISSION, Jenkins.getInstance().hasPermission(CredentialsProvider.DELETE),
-                CREDENTIAL_UPDATE_PERMISSION, Jenkins.getInstance().hasPermission(CredentialsProvider.UPDATE),
-                CREDENTIAL_MANAGE_DOMAINS_PERMISSION, Jenkins.getInstance().hasPermission(CredentialsProvider.MANAGE_DOMAINS)
+                CREDENTIAL_CREATE_PERMISSION, orgBase.hasPermission(CredentialsProvider.CREATE),
+                CREDENTIAL_VIEW_PERMISSION, orgBase.hasPermission(CredentialsProvider.VIEW),
+                CREDENTIAL_DELETE_PERMISSION, orgBase.hasPermission(CredentialsProvider.DELETE),
+                CREDENTIAL_UPDATE_PERMISSION, orgBase.hasPermission(CredentialsProvider.UPDATE),
+                CREDENTIAL_MANAGE_DOMAINS_PERMISSION, orgBase.hasPermission(CredentialsProvider.MANAGE_DOMAINS)
         );
     }
 
     private boolean isAnonymous(String name){
         return name.equals("anonymous") || user.getId().equals("anonymous");
     }
+
+    private static final Pattern pattern = Pattern.compile("/blue/organizations/([^/]*)/");
+
+    private void initOrganizationBase() {
```

Comment:
Matter of style but this method should return `AccessControlled` and then set orgBase in the constructor. This is more clear reading the code as what this function does instead of setting some state thru its logic.